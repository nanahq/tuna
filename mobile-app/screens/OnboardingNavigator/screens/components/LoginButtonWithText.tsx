import {StyleProp, Text, TextStyle, TouchableOpacity} from "react-native";
import {tailwind} from "@tailwind";
import {NavigationProp, useNavigation} from "@react-navigation/native";
import {OnboardingParamsList} from "@screens/OnboardingNavigator/OnboardingNav";
import {OnboardingScreenName} from "@screens/OnboardingNavigator/ScreenName.enum";

export function LoginButtonWithText (props: {style?: StyleProp<TextStyle>}): JSX.Element {
    const navigation  = useNavigation<NavigationProp<OnboardingParamsList>>()

    function goToLogin (): void {
        navigation.navigate(OnboardingScreenName.LOGIN)
    }
    return (
        <TouchableOpacity onPress={goToLogin} testID="LoginButtonWithText.Button" style={tailwind(' mt-8')}>
            <Text testID="LoginButtonWithText.Text" style={[tailwind('text-white font-normal text-sm text-center'), props.style]}>Do you have an account? Login</Text>
        </TouchableOpacity>
    )
}


export function LogoutButtonWithText (props: {style?: StyleProp<TextStyle>}): JSX.Element {
    const navigation  = useNavigation<NavigationProp<OnboardingParamsList>>()

    function goToLogin (): void {
        navigation.navigate(OnboardingScreenName.SIGN_UP_PROFILE)
    }
    return (
        <TouchableOpacity onPress={goToLogin} testID="LoginButtonWithText.Button" style={tailwind(' mt-8')}>
            <Text testID="LoginButtonWithText.Text" style={[tailwind('text-white font-normal text-sm text-center'), props.style]}> You don't have an account? Sign up</Text>
        </TouchableOpacity>
    )
}
