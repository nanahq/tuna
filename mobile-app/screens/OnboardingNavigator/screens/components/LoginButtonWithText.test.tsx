import {render} from '@testing-library/react-native'
import {LoginButtonWithText} from './LoginButtonWithText'

jest.mock("@react-navigation/native", () => ({
    useNavigation: jest.fn(),
}));

describe('<LoginButtonWithText />', () => {
    it('should match snapshot', () => {

        const rendered = render (
            <LoginButtonWithText  />
        )

        // @ts-ignore
        expect(rendered.toJSON()).toMatchSnapshot()

    })

})
