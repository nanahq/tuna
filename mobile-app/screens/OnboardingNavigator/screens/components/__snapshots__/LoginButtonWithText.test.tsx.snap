// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButtonWithText /> should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "marginTop": 32,
      "opacity": 1,
    }
  }
  testID="LoginButtonWithText.Button"
>
  <Text
    style={
      {
        "color": "rgba(255, 255, 255, 1)",
        "fontFamily": "RegularFont",
        "fontSize": 14,
        "fontWeight": "400",
        "lineHeight": 20,
        "textAlign": "center",
      }
    }
    testID="LoginButtonWithText.Text"
  >
    Do you have an account? Login
  </Text>
</View>
`;
